src/annotation_processor.md

```diff
@@ -0,0 +1,149 @@
+
+Here are examples:
+Example case 1: Users might want the annotation of a field be inherited. The fields can reuse the annotations from upper level.
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, we might find both `visitedUrls` and `visitedUserProfiles` have `persistencePolicyInDays` annotated as 30.
+//<inheritance example>
+
+Example case 2: Users might want the annotation of a field be overriden. Override is where the annotation on a field or a schema might get updated when other annotations assign it another value.
+//<override example>
+```pdl
+@persistencePolicyInDays = 30
+record  UserVisitRecord {
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  visitedUserProfiles: array[UserRecord]
+}
+```
+Reading from the schema, a user might find the field `visitedUserProfiles`s `persistencePolicyInDays` annotation is 30 but `visitedUrls`'s `persistencePolicyInDays` annotation has value 10, which overrides the original value inherited(which was 30).
+
+
+Example case 3:  Override might also happen when some annotation needs to be updated by another value assigned from another annotation location
+```pdl
+record UserVisitRecord {
+  //...
+  @persistencePolicyInDays = 365
+  recycledChatHistories: array[chat]
+
+  @persistencePolicyInDays = 10
+  visitedUrls: array[URL]
+  //...
+}
+
+record EnterpriseUserRecord {
+  userName: UserName
+  //...
+  @persistencePolicyInDays = {"/recycledChatHistories" : 3650}
+  visitRecord: UserVisitRecord
+  //...
+}
+```
+In this example, schema `EnterpriseUserRecord` reused `UserVisitRecord` and its annotation in the field `visitRecord`, but it overrides the annotation value of `recycledChatHistories`.
+
+
+All above examples shows that inheritance and overrides gives more extensibility to the schema annotations. Users should be free to define their own rules regarding how those annotations can be read. There should be an imambigous annotation value for the fields after user's rules are applied. We call this processing step to figure out the eventual value of a field or schema's annotation "resolution" and we call such value "resolved" value.
+
+What is more, users could have their own customized logic to process the annotations and define the custom behavior when annotations is overriden or inheritted, and even have the flexibility to set a customized resolved value by calling another local function or remote procedure.  The need for such extensibility gives the motivation for Annotation Processor. We aim to build a tool that can process annotations during schema resolution, process all interested annotations by plugin-ing user's own logic and save the "resolved" value back to corresponding DataSchema.
```

Comment:
nits:
- `inheritted` -> `inherited`
- First sentence is a run-on sentence.
- Extra space at the beginning of the second sentence.
- `interested annotations` -> `relevant annotations` or `applicable annotations`
- `plugin-ing` -> `plugging in`
- `user's own logic` -> `the user's logic`
- `and save the` -> `and saving the`
- `back to corresponding DataSchema` -> `back to the corresponding DataSchema`